src/Microsoft.Health.Fhir.Core/Features/Operations/BulkDelete/Mediator/CreateBulkDeleteHandler.cs

```diff
@@ -0,0 +1,80 @@
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.BulkDelete.Mediator
+{
+    public class CreateBulkDeleteHandler : IRequestHandler<CreateBulkDeleteRequest, CreateBulkDeleteResponse>
+    {
+        private IAuthorizationService<DataActions> _authorizationService;
+        private IQueueClient _queueClient;
+        private RequestContextAccessor<IFhirRequestContext> _contextAccessor;
+        private ISearchService _searchService;
+
+        public CreateBulkDeleteHandler(
+            IAuthorizationService<DataActions> authorizationService,
+            IQueueClient queueClient,
+            RequestContextAccessor<IFhirRequestContext> contextAccessor,
+            ISearchService searchService)
+        {
+            _authorizationService = authorizationService;
+            _queueClient = queueClient;
+            _contextAccessor = contextAccessor;
+            _searchService = searchService;
+        }
+
+        public async Task<CreateBulkDeleteResponse> Handle(CreateBulkDeleteRequest request, CancellationToken cancellationToken)
+        {
+            DataActions requiredDataAction = request.DeleteOperation == DeleteOperation.SoftDelete ? DataActions.Delete : DataActions.HardDelete | DataActions.Delete;
+            if (await _authorizationService.CheckAccess(requiredDataAction, cancellationToken) != requiredDataAction)
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            var searchParameters = new List<Tuple<string, string>>(request.ConditionalParameters); // remove read only restriction
+            var dateCurrent = new PartialDateTime(Clock.UtcNow);
+            searchParameters.Add(Tuple.Create("_lastUpdated", $"lt{dateCurrent}"));
+
+            // Should not run bulk delete if any of the search parameters are invalid as it can lead to unpredicatable results
+            await _searchService.ConditionalSearchAsync(request.ResourceType, searchParameters, -1, cancellationToken);
+            if (_contextAccessor.RequestContext?.BundleIssues?.Count > 0)
+            {
+                throw new BadRequestException(_contextAccessor.RequestContext.BundleIssues.Select(issue => issue.Diagnostics).ToList());
+            }
+
+            var definitions = new List<string>();
+            var processingDefinition = new BulkDeleteDefinition(JobType.BulkDeleteOrchestrator, request.DeleteOperation, request.ResourceType, searchParameters, _contextAccessor.RequestContext.Uri.ToString(), _contextAccessor.RequestContext.BaseUri.ToString());
+            definitions.Add(JsonConvert.SerializeObject(processingDefinition));
+            var jobInfo = await _queueClient.EnqueueAsync((byte)QueueType.BulkDelete, definitions.ToArray(), null, false, false, cancellationToken);
+
+            // Bad, fix this to be better
+            if (jobInfo.Count < 1)
+            {
```

Comment:
For this condition, should we add a more detailed comment? A work-item id? More details to the exception mentioning that there was not Job Info available? Something that would make DRI's life easier.

Reply:
Fixed the code and removed comment.